src/context/appContext.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+import { Users, useUserLazyQuery } from '../generated/graphql';
+import useLocalStorage from '../hooks/useLocalStorage';
+
+type AppContextValue = {
+  user: Users | undefined;
+  isLoginModalOpen: boolean;
+  setUser: React.Dispatch<React.SetStateAction<Users | undefined>>;
+  setIsLoginModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+type AppContextProviderProps = {
+  children: ReactNode;
+  testUser?: Users;
+};
+
+const AppContext = createContext<AppContextValue>({
+  user: undefined,
+  isLoginModalOpen: false,
```

Comment:
- If your routes are protected by the login state, this doesn't need to live here.
- I mentioned this before but when you build out contexts you want to hide implementation details. For example instead of exposing `setUser`, You should expose a `login` and `logout` function. The person calling that method doesn't need to know really how it's built but rather they just need to expect that the right thing will happen when they call it